client/src/util/renderThrottle.ts

```diff
@@ -1,17 +1,16 @@
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-export default function renderThrottle(callback: any) {
+export default function renderThrottle(
+  callback: () => void
+): (this: unknown) => void {
```

Comment:
I think the return should be a function that accepts arguments for the callback instead of `this`, since `this` is the execution context, not the arguments!

```ts
// lodash example
function throttle(fn: (...args: any[]) => void, delay: number): (...args: any[]) => void;
```

I just found that TS now has `Parameters` util! https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype

So we can do something like:

```ts
function throttle<T extends (...args: never[]) => void>(
  callback: T,
  timeoutMS: number
): (...args: Parameters<T>) => void { ... }
```

Inspiration: https://gist.github.com/ca0v/73a31f57b397606c9813472f7493a940

Reply:
why add all this complexity and change if the current code suffices?  Using `this` to pass context is a very old tradition, and is equally good in this use case.

No issues with using explicit args, but it is change in behavior introducing more risk & code motion to this "typing" PR.